windows/RNSound/RNSound/RNSound.cpp

```diff
@@ -0,0 +1,215 @@
+#include "pch.h"
+#include "RNSound.h"
+
+using namespace winrt::Windows::ApplicationModel;
+using namespace winrt::Windows::Media::Core;
+using namespace winrt::Windows::Media::Playback;
+using namespace winrt::Windows::Storage;
+using namespace winrt::RNSound;
+
+void RNSound::Prepare(const std::string& fileName, int key, JSValueObject options,
+    std::function<void(JSValue, JSValue)> callback) noexcept {
+    bool enableSMTCIntegration = true;
+    if (options["enableSMTCIntegration"] != nullptr) {
+        enableSMTCIntegration = options["enableSMTCIntegration"].AsBoolean();
+    }
+
+    MediaPlayer player{};
+    loadFile(player, fileName);
+
+    player.CommandManager().IsEnabled(enableSMTCIntegration);
+
+    if (player == nullptr) {
+        JSValueObject e{};
+        e["code"] = -1;
+        e["message"] = "resource not found";
+        callback(std::move(e), nullptr);
+        return;
+    }
+
+    player.AutoPlay(false);
+    player.MediaOpened(
+        [callback](winrt::Windows::Media::Playback::MediaPlayer sender,
+               const winrt::Windows::Foundation::IInspectable &) {
+            JSValueObject props{};
+            props["duration"] = sender.PlaybackSession().NaturalDuration().count() / 
+                winrt::impl::filetime_period::den;
+            callback(nullptr, std::move(props));
+        });
+
+    playerPool[key] = player;
+}
+
+winrt::Windows::Foundation::IAsyncAction RNSound::loadFile(MediaPlayer player,
+    const std::string &fileName) {
+    StorageFile file{nullptr};
```

Comment:
this change seems to break loading a file from url. All thats needed is the following before looking for storage folders/storage files:

    if (fileName.rfind("http", 0) == 0)
    {
        winrt::Windows::Foundation::Uri uri{ winrt::to_hstring(fileName) };
        auto mediaSource = MediaSource::CreateFromUri(uri);
        player.Source(mediaSource);
    }

Reply:
I have some changes that would support doing this, but the janeasystems repo doesn't seem to support pull requests from additional contributors.

Reply:
Ah thanks for catching that! Let me check and get back to you to see what the best option we have for adding changes to that repo is.

Reply:
Hi @chrisfromwork , what error or message are you getting when trying to create a pull request? I think it should be possible to open a PR.

Reply:
Changes from @chrisfromwork are now in the PR.

Reply:
@jamesreggio I do not see the JameaSystems repo as a repo I can target with a cross fork pr. I am not sure why

Reply:
thanks @chiaramooney and @jaimecbernardo for picking up this uri support change